src/go/k8s/pkg/console/admin.go

```diff
@@ -0,0 +1,76 @@
+package console
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources"
+	"github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources/certmanager"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/sasl/scram"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	adminutils "github.com/redpanda-data/redpanda/src/go/k8s/pkg/admin"
+)
+
+// NewAdminAPI create an Admin API client
+func NewAdminAPI(ctx context.Context, cl client.Client, scheme *runtime.Scheme, cluster *redpandav1alpha1.Cluster, clusterDomain string, adminAPI adminutils.AdminAPIClientFactory, log logr.Logger) (adminutils.AdminAPIClient, error) {
+	headlessSvc := resources.NewHeadlessService(cl, cluster, scheme, nil, log)
+	clusterSvc := resources.NewClusterService(cl, cluster, scheme, nil, log)
+	pki := certmanager.NewPki(
+		cl,
+		cluster,
+		headlessSvc.HeadlessServiceFQDN(clusterDomain),
+		clusterSvc.ServiceFQDN(clusterDomain),
+		scheme,
+		log,
+	)
+	adminTLSConfigProvider := pki.AdminAPIConfigProvider()
+	adminAPIClient, err := adminAPI(
+		ctx,
+		cl,
+		cluster,
+		headlessSvc.HeadlessServiceFQDN(clusterDomain),
+		adminTLSConfigProvider,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating AdminAPIClient: %w", err)
+	}
+	return adminAPIClient, nil
+}
+
+// NewKafkaAdminClient create a franz-go admin client
+func NewKafkaAdmin(ctx context.Context, cl client.Client, cluster *redpandav1alpha1.Cluster) (*kadm.Client, error) {
+	opts := []kgo.Opt{kgo.SeedBrokers(getBrokers(cluster)...)}
+	if cluster.Spec.EnableSASL {
+		// Use Pandaproxy superuser to manage Kafka
+		// Console Kafka Service Account can't add ACLs to itself
+		pandaProxySu := types.NamespacedName{
```

Comment:
I think so. If user set `spec.enableSASL=false` Although we are just using pandaproxy here if SASL is enabled. If user decided to set `enableSASL=false` I think we should be able to connect to Kafka Admin without setting SASL config.

Reply:
do console talk to kafka through pandaproxy? Or why are we using pandaproxy SU? what if we're interfacing with cluster that has proxy disabled so no superuser?

Reply:
There's a thread talking about future authz for Console -> Redpanda [REF](https://redpandadata.slack.com/archives/C01H6JRQX1S/p1661178370034779) I think this needs more discussion. I will use the superuser (i.e. <cluster-name>-superuser) instead of pandaproxy at the mean time.